.github/workflows/build-test-osx-x86.yaml

```diff
@@ -78,20 +80,17 @@ jobs:
       # hit the cache unfortunately, but we would spend time only for installing
       # those new packages (ideally we would want to regenerate the cache by
       # using an opam.pock in the cache key).
-      - name: Restore Opam Cache
-        id: restore-opam-cache
-        uses: actions/cache/restore@v3
+      - name: Cache Opam
+        uses: actions/cache@v3
         if: ${{ inputs.use-cache }}
-        # The size of the tgz of ~/.opam is around 500MB, which usually takes less than
-        # 2min to download from GHA. If it takes more than that, that means something
-        # wrong is going on in which case it's better to timeout early than spend
-        # 10min (the default timeout) and then still timeout.
         env:
```

Comment:
Why removing that comment?
I wrote that comment because PA people don't have a deep understanding of GHA but according to our devops philosophy, PA people are supposed to maintain/understand/modify those files (with some help from infra when we need to do new/tricky things), so we need to keep those comments.